papaya-web/src/main/java/pl/edu/agh/papaya/rest/jira/service/JiraRestService.java

```diff
@@ -0,0 +1,185 @@
+
+            JiraBoardDto jiraBoardDto = new JiraBoardDto()
+                    .boardName(board.getString(NAME))
+                    .boardId(board.getLong(ID))
+                    .boardType(board.getString(TYPE))
+                    .projectId(boardProjectDetails.getLong(PROJECT_ID))
+                    .projectKey(boardProjectDetails.getString(PROJECT_KEY))
+                    .projectName(boardProjectDetails.getString(PROJECT_NAME));
+
+            jiraBoardDtos.add(jiraBoardDto);
+        }
+        return jiraBoardDtos;
+    }
+
+    private void checkFullJiraConfig(Project project) {
+        checkJiraUrlConfig(project);
+        checkJiraRequestTokenConfig(project);
+        if (project.getJiraSecret() == null || project.getAccessToken() == null) {
+            throw new NotAcceptableException("Missing Jira configuration");
+        }
+    }
+
+    private void checkJiraRequestTokenConfig(Project project) {
+        if (project.getRequestToken() == null) {
+            throw new NotAcceptableException("Missing Jira request token configuration");
+        }
+    }
+
+    public void setJiraSecret(Project project, String jiraSecret) {
+        checkJiraUrlConfig(project);
+        checkJiraRequestTokenConfig(project);
+
+        projectService.updateJiraSecret(project, jiraSecret);
+        String accessToken = projectService.getJiraUrl(project)
+                .flatMap(url -> jiraAuthorizationService.getAccessToken(url, jiraSecret, project.getRequestToken()))
+                .orElseThrow(() -> new NotAcceptableException("Incorrect Jira configuration"));
+
+        projectService.updateAccessToken(project, accessToken);
+    }
+
+    public void setJiraBoard(Project project, JiraBoardDto jiraBoardDto) {
+        checkFullJiraConfig(project);
+
+        projectService.updateJiraBoardId(project, jiraBoardDto.getBoardId());
+        projectService.updateJiraProjectId(project, jiraBoardDto.getProjectId());
+    }
+
+    public List<JiraSprintDto> getAvailableJiraSprints(Sprint sprint) {
+        Project project = sprint.getProject();
+        JSONArray jiraSprints = projectService.getJiraUrl(project)
+                .flatMap(url -> jiraAuthorizationService.makeAuthorizedGetRequest(url, project.getJiraSecret(),
+                        project.getAccessToken(), String.format(JIRA_SPRINTS_ENDPOINT, project.getJiraBoardId())))
+                .orElseThrow(BadRequestException::new)
+                .getJSONObject(0)
+                .getJSONArray(VALUES);
+
+        List<JiraSprintDto> jiraSprintDtos = new LinkedList<>();
+        for (int i = 0; i < jiraSprints.length(); i++) {
+            JSONObject sprintJsonObject = jiraSprints.getJSONObject(i);
+            LocalDateTime jiraSprintStart = convertLocalDateTimeFromUtc(sprintJsonObject.getString(START_DATE));
+            if (Duration.between(jiraSprintStart, sprint.getDurationPeriod().getStart()).toMinutes()
+                    <= MAX_SPRINT_OFFSET) {
```

Comment:
Tego jeszcze nie testowałam, ale skoro skleiłam strefy czasowe taśmą, to powinni być okej